cl/recap/mergers.py

```diff
@@ -843,6 +845,24 @@ async def add_docket_entries(
             for tag in tags:
                 tag.tag_object(rd)
 
+        attachments = docket_entry.get("attachments")
+        if attachments is not None:
+            task = asyncio.create_task(
+                merge_attachment_page_data(
+                    d.court,
+                    d.pacer_case_id,
+                    rd.pacer_doc_id,
+                    docket_entry["document_number"],
+                    None,
+                    attachments,
+                    False,
+                )
+            )
+            attachment_tasks.add(task)
+
+    if attachment_tasks:
+        await asyncio.wait(attachment_tasks)
```

Comment:
It seems there is an issue around here. The test appears to work properly, but when I performed the same docket upload using CURL, the `attachment_tasks` got stuck. Attachments are not added, and the request doesn't receive a response.

I also got this error:

> Application instance <Task pending name='Task-39' coro=<ASGIStaticFilesHandler.__call__() running at /usr/local/lib/python3.11/site-packages/django/contrib/staticfiles/handlers.py:101> wait_for=<Future pending cb=[_chain_future.<locals>._call_check_cancel() at /usr/local/lib/python3.11/asyncio/futures.py:387, Task.task_wakeup()]>> for connection <WebRequest at 0xffff7f9229d0 method=POST uri=/api/rest/v3/recap/ clientproto=HTTP/1.1> took too long to shut down and was killed.

Reply:
switched to non-task based approach, does that seem to work ok?

Reply:
> when I performed the same docket upload using CURL, the `attachment_tasks` got stuck.

I'm suspecting that this likely has something to do with passing the same court model object instance to multiple tasks. I'll experiment with refactoring things there a bit after the non-task version is merged to avoid reusing the court model object instances like that.

Reply:
Great! the non-task approach is working properly.

Reply:
> It seems there is an issue around here. The test appears to work properly, but when I performed the same docket upload using CURL, the `attachment_tasks` got stuck. Attachments are not added, and the request doesn't receive a response.
> 
> I also got this error:
> 
> > Application instance <Task pending name='Task-39' coro=<ASGIStaticFilesHandler.**call**() running at /usr/local/lib/python3.11/site-packages/django/contrib/staticfiles/handlers.py:101> wait_for=<Future pending cb=[_chain_future.._call_check_cancel() at /usr/local/lib/python3.11/asyncio/futures.py:387, Task.task_wakeup()]>> for connection <WebRequest at 0xffff7f9229d0 method=POST uri=/api/rest/v3/recap/ clientproto=HTTP/1.1> took too long to shut down and was killed.

I think this is actually due to [this asgiref bug](https://github.com/django/asgiref/issues/348).